proptest/src/collection.rs

```diff
@@ -597,90 +600,40 @@ impl<T: ValueTree> ValueTree for VecValueTree<T> {
     type Value = Vec<T::Value>;
 
     fn current(&self) -> Vec<T::Value> {
-        self.elements
-            .iter()
-            .enumerate()
-            .filter(|&(ix, _)| self.included_elements.test(ix))
-            .map(|(_, element)| element.current())
-            .collect()
+        self.current.as_ref().replace(vec![])
```

Comment:
I don't understand what this is doing. Can you please add a comment?

Reply:
Will do! TL;DR is that this allows us to output the the vector without copying it out. That's better for performance because original implementation and copying both need to iterate the vector twice.

Reply:
RC enables interior mut even though we only have `&self`, and RefCell has a replace API we can use to replace the contents and get back the current content. Combine both to get zero-copy values out of a static pointer.

Reply:
32ce7073475692638acfc594ebb94aa94d82c1d5